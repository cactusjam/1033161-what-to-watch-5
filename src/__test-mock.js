import {NameSpace} from "./store/reducers/root-reducer.js";

export const TEST_MOCK_MOVIE = {
  id: 1,
  genre: ``,
  title: ``,
  cover: ``,
  poster: ``,
  background: ``,
  backgroundColor: ``,
  releaseYear: 1,
  promo: ``,
  video: ``,
  duration: 1,
  rating: {
    score: 1,
    countOfVotes: 1
  },
  description: ``,
  director: ``,
  starring: [],
  isFavorite: true,
};

export const TEST_MOCK_COMMENT = {
  id: 1,
  author: ``,
  rating: 22,
  reviewText: ``,
  date: ``
};

export const TEST_MOCKS = {
  id: 1,
  noop: () => {},
  match: {
    params: {
      id: `1`,
    },
  },
  movie: TEST_MOCK_MOVIE,
  movies: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
  genresFilter: [`All genres`, ``, ``, ``, ``, ``],
  genreFilter: `All genres`,
  review: TEST_MOCK_COMMENT,
  userData: {
    id: 1,
    email: ``,
    name: ``,
    authorizationStatus: ``,
    userAvatar: ``,
  },
  reviews: [TEST_MOCK_COMMENT, TEST_MOCK_COMMENT, TEST_MOCK_COMMENT],
  authorizationStatusTrue: `AUTH`,
  authorizationStatusFalse: `NO_AUTH`,
};

export const TEST_MOCK_STORE = {
  [NameSpace.USER]: {
    id: 1,
    email: ``,
    authorizationStatus: `AUTH`,
    userAvatar: ``,
  },
  [NameSpace.GENRES]: {
    activeFilterGenre: `Adventure`
  },
  [NameSpace.DATA]: {
    movies: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
    reviews: [],
    currentMovie: TEST_MOCK_MOVIE,
    promoMovie: TEST_MOCK_MOVIE,
    favorites: [TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE, TEST_MOCK_MOVIE],
    defaultFilmsCount: 8,
    isDataSending: false,
    isDataSendError: false
  },
};
