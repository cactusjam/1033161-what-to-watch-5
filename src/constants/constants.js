const TabName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const GenresFilter = {
  ALL: `All genres`,
  COMEDY: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMA: `Dramas`,
  HORROR: `Horror`,
  KIDS: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLER: `Thrillers`,
};

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const FILMS_COUNT_PER_CLICK = 8;

export const Rating = {
  BAD: 3,
  NORMAL: 5,
  GOOD: 8,
  VERY_GOOD: 10,
};

export const AppRoute = {
  LOGIN: `/login`,
  RESULT: `/result`,
  MOVIES: `/films`,
  MOVIES_PROMO: `/films/promo`,
  MY_LIST: `/mylist`,
  REVIEWS: `/comments`,
  CURRENT_REVIEW: `/films/:id/review`,
  CURRENT_MOVIE: `/films/:id`,
  CURRENT_PLAYER: `/player/:id`,
  FAVORITE: `/favorite`,
};

export const HttpCode = {
  OK: 200,
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  SERVER_ERROR: 500,
};

export {TabName, GenresFilter, AuthorizationStatus, FILMS_COUNT_PER_CLICK};
