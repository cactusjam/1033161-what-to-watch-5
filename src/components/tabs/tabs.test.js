import React from "react";
import renderer from "react-test-renderer";
import Tabs from "./tabs";
import {TEST_MOCKS} from "../../__test-mock.js";
import {TabName} from "../../constants/constants";


describe(`Tabs`, () => {
  it(`Should Tabs render correctly`, () => {
    const tree = renderer
    .create(
        <Tabs
          activeTab={TabName.OVERVIEW}
          onActiveMovieChange={TEST_MOCKS.noop}
          movie={TEST_MOCKS.movie}
          reviews={TEST_MOCKS.reviews}
        />
    )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
