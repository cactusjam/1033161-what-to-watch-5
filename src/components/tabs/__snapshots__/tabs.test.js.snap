// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Should Tabs render correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      1
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        1
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        
      </strong>
    </p>
  </div>
</div>
`;
